notifications_utils/postal_address.py

```diff
@@ -163,3 +163,9 @@ def format_postcode_for_printing(postcode):
     elif "BFPO" in postcode:
         return postcode[:4] + " " + postcode[4:]
     return postcode[:-3] + " " + postcode[-3:]
+
+
+@lru_cache(maxsize=8)
```

Comment:
Optional: Add to the commit message or code why maxsize 8 was chosen. You say `When validating an address we look at the postcode multiple times.` so I'm guessing we look at it up to something like 8 times so that's why it was picked.

Out of interest, when processing a CSV why do we look at the postcode multiple times?

Reply:
```suggestion
# When processing an address we look at the postcode twice when
# normalising it, and once when validating it. So 8 is chosen because
# it’s 3, doubled to give some headroom then rounded up to the nearest
# power of 2
@lru_cache(maxsize=8)
```